rfd/0167-console-service.md

```diff
@@ -0,0 +1,255 @@
+
+#### Endpoints
+
+In addition to the `pprof` endpoints, the service will also have endpoints that
+will be used to retrieve and set the applications's log level:
+- `PUT /log-level` will receive the new log level on its body as text. The
+  the level will be parsed using `UnmarshalText` from `slog.Level`, meaning the
+  the provided level must follow the `slog` format.
+- `GET /log-level` will retrieve the application's current log level. This will
+  be done by returning the value present on the `slog.LevelVar`.
+
+The log level change will consist on updating `slog` log level and `logrus`
+logger (legacy):
+- `slog`: Pass a `slog.LevelVar` to the handler. The `slog.LevelVar` is then
+   stored where the endpoint can modify it.
+- `logrus`: Both `Config` and default logger will need to be updated using
+  `SetLevel`.
+
+### `teleport console` command
+
+A new set of commands will be introduced to `teleport` to consume the new
+service.
+
+Commands will have a common argument for receiving the instance configuration.
+This allows loading the configuration to locate the data directory, and later
+read the Node ID for generating the socket name.
+
+#### Connecting to the server
+
+Since the service will listen into a Unix domain socket. There will be a few
+differences while initializing/configuring the `http.Client`. The client will
+need to use a custom transport, which will dial to the socket.
+
+<details>
+<summary>Example of HTTP Client/Server connecting through Unix domain socket<summary>
+
+```go
+// Server will look like a regular HTTP server, the only difference is that the
+// listener will use "unix" network instead of "tcp".
+func startServer(socketAddr string) {
+	listener, _ := net.Listen("unix", socketAddr)
+
+	// Setup the http server with mux.
+	mux := http.NewServeMux()
+	// mux.HandleFunc(...)
+
+	server := http.Server{Handler: mux}
+	server.Serve(listener)
+}
+
+// The client however, requires a different transport. The differences is on the
+// DialContext implementation.
+func createClient(socketAddr string) *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socketAddr)
+			},
+		}
+	}
+}
+
+// While making requests using the client, the users will still require to
+// provide the full URL since the client always validate them.
+res, err := client.Get("http://console-service/debug/pprof/heap")
+```
+</details>
+
+#### Changing log level command
+
+Changes the application log level.
+
+`teleport [-c config-path] console set-log-level [LEVEL]`
+
+|Flag|Description|Default value|
+|----|-----------|-------------|
+|`-c,--config`|Teleport configuration path.|`/etc/teleport.yaml`|
+|`LEVEL`|Log level (case-insensitive). Any of: `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`|Required|
+
+Usage examples:
+```bash
+$ teleport console set-log-level DEBUG
+$ teleport -c /random/teleport.yaml console set-log-level INFO
```

Comment:
could you also share the sample output? Ideally we also show the previous level they are changing from.